### PR TITLE
Industry mapping cms

### PR DESCRIPTION
**Pull request by FarazA22**
### Ticket

https://www.pivotaltracker.com/story/show/186352406
## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: web/public/mgmt/config.yml [hunk 1]
```diff
@@ -2995,6 +2995,32 @@ collections:
       - { label: "Internal Notes", name: "notesMd", widget: "markdown", required: false }
       - { label: "Filename (disabled after file creation)", name: "filename", widget: "write-once-read-only" }
       - { label: "Name/Title", name: "name", widget: "string" }
+      - {
+          label: "Apply to all - overrides industry and sector dropdown",
```
Comment (AnJuHyppolite) [inline web/public/mgmt/config.yml:2999]: [dust] Capitalize some of these labels: change to `"Apply to All - Overrides Industry and Sector Dropdown"`

File: web/public/mgmt/config.yml [hunk 2]
```diff
@@ -2995,6 +2995,32 @@ collections:
...
+          name: "applyToAllUsers",
+          widget: "boolean",
+          required: false,
+        }
+      - label: Apply to industries
```
Comment (AnJuHyppolite) [inline web/public/mgmt/config.yml:3004]: [dust] Capitalize some of these labels: change to `"Apply to Industries"`

File: web/public/mgmt/config.yml [hunk 3]
```diff
@@ -2995,6 +2995,32 @@ collections:
...
+        name: industryIds
+        widget: relation
+        multiple: true
+        required: false
+        default: []
+        collection: "mappings"
+        search_fields: ["industries.*.name"]
+        value_field: "industries.*.id"
+        display_fields: ["industries.*.name"]
+      - label: Apply to sectors
```
Comment (AnJuHyppolite) [inline web/public/mgmt/config.yml:3014]: [dust] Capitalize some of these labels: change to `"Apply to Sectors"`

File: web/public/mgmt/config.yml [hunk 4]
```diff
@@ -3025,6 +3051,32 @@ collections:
       - { label: "Internal Notes", name: "notesMd", widget: "markdown", required: false }
       - { label: "Filename (disabled after file creation)", name: "filename", widget: "write-once-read-only" }
       - { label: "Name/Title", name: "name", widget: "string" }
+      - {
+          label: "Apply to all - overrides industry and sector dropdown",
```
Comment (AnJuHyppolite) [inline web/public/mgmt/config.yml:3055]: [dust] Capitalize some of these labels: change to "Apply to All - Overrides Industry and Sector Dropdown"

File: web/public/mgmt/config.yml [hunk 5]
```diff
@@ -3025,6 +3051,32 @@ collections:
...
+          name: "applyToAllUsers",
+          widget: "boolean",
+          required: false,
+        }
+      - label: Apply to industries
```
Comment (AnJuHyppolite) [inline web/public/mgmt/config.yml:3060]: [dust] Capitalize some of these labels: change to `"Apply to Industries"`

File: web/public/mgmt/config.yml [hunk 6]
```diff
@@ -3025,6 +3051,32 @@ collections:
...
+        name: industryIds
+        widget: relation
+        multiple: true
+        required: false
+        default: []
+        collection: "mappings"
+        search_fields: ["industries.*.name"]
+        value_field: "industries.*.id"
+        display_fields: ["industries.*.name"]
+      - label: Apply to sectors
```
Comment (AnJuHyppolite) [inline web/public/mgmt/config.yml:3070]: [dust] Capitalize some of these labels: change to `"Apply to Sectors"`

File: web/public/mgmt/config.yml (hunk 1 above)
Comment (FarazA22) [reply web/public/mgmt/config.yml:2999]: Hey Anju, I looked into the CMS and noticed all the characters are capitalized in the CMS; based on that, I'm going to leave it as it.

Comment (FarazA22) [reply web/public/mgmt/config.yml:2999]: Screenshot 2024-01-17 at 9 17 45 AM

File: web/src/components/dashboard/QuickActionContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  const sectorId = business?.profileData.sectorId;
+  const isDesktopAndUp = useMediaQuery(MediaQueries.desktopAndUp);
+
+  if (!business) {
+    return <></>;
+  }
+  const findMatch = (action: QuickActionTask | QuickActionLink): boolean => {
+    let result = false;
+    if (!!action.industryIds && !!industryId && action.industryIds.includes(industryId)) result = true;
+    if (!!action.sectorIds && !!sectorId && action.sectorIds.includes(sectorId)) result = true;
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/components/dashboard/QuickActionContainer.tsx:24]: you're right, the `result` variable is a tactic from the refactoring book.

File: web/src/components/dashboard/QuickActionContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    if (action.applyToAllUsers) result = true;
+    return result;
+  };
+
+  const filteredTasks = quickActionTasks
+    ? quickActionTasks
+        .filter((action) => findMatch(action))
+        .map((action) => <QuickActionTile type="task" quickAction={action} key={action.filename} />)
+    : [];
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/components/dashboard/QuickActionContainer.tsx:33]: It was blowing up when the page initially loaded, so I added it in. You may want to look into it.

File: web/src/components/dashboard/QuickActionContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+  const filteredLinks = quickActionLinks
+    ? quickActionLinks
+        .filter((action) => findMatch(action))
+        .map((action) => <QuickActionTile type="link" quickAction={action} key={action.filename} />)
+    : [];
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/components/dashboard/QuickActionContainer.tsx:39]: It was blowing up when the page initially loaded, so I added it in. You may want to look into it.

File: web/src/lib/types/types.ts [hunk 10]
```diff
@@ -212,10 +212,16 @@ export interface QuickActionTask extends QuickAction {
   callToActionLink: string | undefined;
   callToActionText: string | undefined;
   form: string | undefined;
+  industryIds: string[];
+  sectorIds: string[];
+  applyToAllUsers: boolean;
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/lib/types/types.ts:217]: You are right, needs to be fixed

File: web/src/lib/utils/markdownReader.ts [hunk 11]
```diff
@@ -184,6 +184,9 @@ type QuickActionLinkGrayMatter = {
   name: string;
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/lib/utils/markdownReader.ts:184]: I think they need to be separate... forgot why

**Pull request merged**